compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/KernelDispatch.cpp

```diff
@@ -67,6 +67,95 @@ static LogicalResult setRootConfigForPackPipeline(func::FuncOp entryPointFn,
       IREE::Codegen::DispatchLoweringPassPipeline::None);
 }
 
+template <typename ADDOP, typename MULOP>
+static bool bodyMatcherForMatmulTranspose(Value yieldVal, Block *body) {
+  auto addOp = yieldVal.getDefiningOp<ADDOP>();
+  if (!addOp) return false;
+  auto mulOp = addOp.getOperand(1).template getDefiningOp<MULOP>();
+  if (!mulOp) return false;
+  auto lhsBlockArg = mulOp.getOperand(0).template dyn_cast<BlockArgument>();
+  auto rhsBlockArg = mulOp.getOperand(1).template dyn_cast<BlockArgument>();
+  auto outBlockArg = addOp.getOperand(0).template dyn_cast<BlockArgument>();
+  if (!lhsBlockArg || !rhsBlockArg || !outBlockArg ||
+      lhsBlockArg.getOwner() != body || rhsBlockArg.getOwner() != body ||
+      outBlockArg.getOwner() != body || lhsBlockArg.getArgNumber() != 0 ||
+      rhsBlockArg.getArgNumber() != 1 || outBlockArg.getArgNumber() != 2) {
+    return false;
+  }
+  return true;
+}
```

Comment:
I appreciate you doing a clever thing to figure this out, but I fear this may be brittle as we scale of the codegen. Generally, at this level it is okay to look at what other backends are doing and lift the code :)
I would do something inspired from the cpu backend here
https://github.com/openxla/iree/blob/main/compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp#L1499-L1523
At the same time be careful as CPU might be supporting more transposes then the transpose what you are in this PR

Reply:
Actually, on reviewing further, I dont think you want to support a linalg.generic that is doing a tranpose, I thought you wanted to support the named op, linalg.tranpose_b that uses the contractionOp interface?

Reply:
I think there is a confusion here. THis is checking for a `linalg.generic` that effectively implements `linalg.matmul_transpose_b`. From the input we might not necessarily get a named op. We might get a `generic` op which is implementing the `linalg.matmul_tranpose_b`.